### PR TITLE
Issue #136 - Use custom formatter to get desired state of deployment and service

### PR DESCRIPTION
**Pull request by alexmt**
Closes https://github.com/argoproj/argo-cd/issues/132
Closes https://github.com/argoproj/argo-cd/issues/136

### CODE REVIEW

File: util/ksonnet/ksonnet.go [hunk 1]
```diff
@@ -132,12 +139,63 @@ func (k *ksonnetApp) Show(environment string) ([]*unstructured.Unstructured, err
...
 }
 
+// remarshal checks resource kind and version and re-marshal using corresponding struct custom marshaller. This ensures that expected resource state is formatter same as actual
+// resource state in kubernetes and allows to find differences between actual and target states more accurate.
+func remarshal(obj *unstructured.Unstructured) error {
+	var newObj interface{}
+	switch obj.GetAPIVersion() + ":" + obj.GetKind() {
+	case "apps/v1beta1:Deployment":
+		newObj = &v1beta1.Deployment{}
+		break
```
Comment (jessesuen) [inline util/ksonnet/ksonnet.go]: Style nit: in go, you don't need to stay break after each case.

File: util/ksonnet/ksonnet.go [hunk 2]
```diff
@@ -132,12 +139,63 @@ func (k *ksonnetApp) Show(environment string) ([]*unstructured.Unstructured, err
...
+		}
+		data, err = json.Marshal(newObj)
+		if err != nil {
+			return err
+		}
+		err = json.Unmarshal(data, obj)
+		if err != nil {
+			return err
+		}
+		obj.Object = diff.RemoveMapFields(oldObj, obj.Object)
```
Comment (jessesuen) [inline util/ksonnet/ksonnet.go]: Please document the reason for all the back and forth marshaling/unmarshalling going on here.

**Review by jessesuen — APPROVED** [review]
minor changes requested

File: util/ksonnet/ksonnet.go (hunk 2 above)
Comment (alexmt) [reply util/ksonnet/ksonnet.go]: done

File: util/ksonnet/ksonnet.go (hunk 1 above)
Comment (alexmt) [reply util/ksonnet/ksonnet.go]: thank you! fixed

**Pull request merged**